js/router.js

```diff
@@ -1,15 +1,153 @@
+import $ from 'jquery';
 import { Router } from 'backbone';
+import { getGuid } from './utils';
+import { getPageContainer } from './utils/selectors';
+import UserPage from './views/UserPage';
+import TransactionsPage from './views/TransactionsPage';
+import TemplateOnly from './views/TemplateOnly';
 
 export default class ObRouter extends Router {
-  constructor() {
+  constructor(options = {}) {
+    super(options);
+    this.options = options;
+
+    // Temporarily requiring a dummy 'users' collection. Eventually, we'll
+    // be getting this data via fetches to the server
+    if (!options.usersCl) {
+      throw new Error('Need me some dummy users!');
+    }
+
+    this.usersCl = options.usersCl;
+
+    if (!options.pageNavVw) {
+      throw new Error('Please provide the app\'s pageNavVw.');
+    }
+
+    this.pageNavVw = options.pageNavVw;
+
     const routes = [
-      // ['home/:state(/:searchText)', 'home'],
-      // ['userPage/:userID(/:state)(/:itemHash)(/:skipNSFWmodal)', 'userPage'],
-      // [/^@([^\/]+)(.*)$/, 'userPageViaHandle'],
-      // ['transactions/:state(/:orderID)(/:tabState)', 'transactions'],
-      // ['settings/:state', 'settings'],
+      [/^@([^\/]+)[\/]?([^\/]*)[\/]?([^\/]*)[\/]?([^\/]*)$/, 'userViaHandle'],
+      [/^(Qm[a-zA-Z0-9]+)[\/]?([^\/]*)[\/]?([^\/]*)[\/]?([^\/]*)$/, 'user'],
+      ['transactions', 'transactions'],
+      ['transactions/:tab', 'transactions'],
+      ['*path', 'pageNotFound'],
     ];
 
-    routes.forEach((route) => this.route.apply(this, route));
+    routes.slice(0)
+      .reverse()
+      .forEach((route) => this.route.apply(this, route));
+
+    this.setAddressBarText();
+
+    $(window).on('hashchange', () => {
+      this.setAddressBarText();
+    });
+  }
+
+  setAddressBarText() {
+    if (location.hash.startsWith('#transactions')) {
```

Comment:
So, I ended up going back to making the `transactions` route not have the handle preface, which means we won't be showing it in the address bar, nor allowing it to be accessed via typing it's url into the address bar.

Having worked through it, if we did need to support `@handle/transactions` urls (both showing them in the address bar and giving the bar the ability to route to them), it wouldn't be that difficult to do (the infrastructure is in place).